code/modules/hydroponics/seed_gene_presets.dm

```diff
@@ -0,0 +1,425 @@
+/obj/item/computer_hardware/hard_drive/portable/plantgene
+	disk_name = "plant gene disk"
+	rarity_value = 40
+	spawn_tags = SPAWN_TAG_PLANT_GENE_DISK
+	spawn_blacklisted = TRUE	// Trade beacons or off-ship loot
+	bad_type = /obj/item/computer_hardware/hard_drive/portable/plantgene	// Base type has no gene info
+	price_tag = 50
+	var/trait_info
+	var/preset_genetype
+	var/list/preset_values = list()
+
+/obj/item/computer_hardware/hard_drive/portable/plantgene/Initialize()
+	var/datum/computer_file/binary/plantgene/preloaded_gene = new /datum/computer_file/binary/plantgene
+
+	// Taken from maintshrooms
+	var/list/firstnames = list (
+		"bleak", "bog", "bum", "candy", "coarse", "corpse", "cramp", "club", "demon", "dog", "dung", "felt", "fly", "goblin", "grave", "greasy", "hypoxylon", "jelly", "junk", "icky", "imp", "ling", "lung", "maggot", "monkey", "mottled", "pixie", "poached", "powder", "pterula", "ramularia", "radiant", "rat", "roach", "rock", "scruffy", "serbian", "shaggy", "slimy", "smelly", "smoky", "space", "spider", "spiky", "splash", "stag", "stinky", "tumbling", "undulate", "vagabond", "veiled", "wall", "web"
+		)
+	var/list/secondnames = list(
+		"amanita", "bane", "beacon", "bolete", "bonnet", "bulgar", "cap", "cone", "conocybe", "coral", "clown", "cremini", "crepidotus", "cup", "cushion", "entoloma", "ear", "fungus", "gill", "heart", "horn", "hydnum", "leafspot", "leoninus", "lichen", "micellium", "morsel", "moss", "mushroom", "ori", "oyster", "panellus", "polypore", "panus", "porcini", "porridge", "puffball", "rod", "rot", "russula", "scale", "smut", "spots", "stem", "stool", "tail", "tears", "tongue", "trumpet", "truffle",  "urn", "vase", "wort"
+		)
+
+	var/plant_name = "[pick(firstnames)] [pick(secondnames)]"
+	var/plant_id = rand(100,999)
```

Comment:
What happens if it randomly ends up twice with the same ID for two drives?

Reply:
Nothing. It's purely for name generation.